### DISCUSSION TITLE
Install Error? MacOS 10.15.7 Catalina

### DISCUSSION THREAD

**Discussion by artcore-c** [section83/MacYTDL#44] [Q&A]
Having some trouble getting MacYTDL off and running, keep getting this error message, and also notice in System Preferences/Privacy/Automation there isn't a checkbox for Safari? 

I can see yt-dlp, ffmpeg and ffprobe are installed from zip downloaded in bin so that's good, the alerts worked also notifying as they downloaded. It really looks like a nice UI what I see in the Help booklet, need something to download tutorials for offline viewing because trying to replay parts as I follow every few minutes over my slow mobile connection lags re-buffering and ends up taking a lot longer to see the parts I need... this has been such a problem for me it's almost 4 weeks and I'm only about 2/3 of the way through an hour and a half long tutorial. Had YouTube Premium subscription for one month but discontinued as I cannot download or transfer to Mac and my iPad has a much smaller viewing area. Older Mac 2014 screen sharing is not impossible but problematic so ideally watching on my desktop from save files was my best hope. I wish there was a way to download faster to help compensate for slow connectivity like I'm able to do with Download Shuttle that segments into 5 parts and won't drop and restart if interrupted.
hope you can help! Thank you and very kind regards. RTH

![Not Found](https://user-images.githubusercontent.com/1493547/141676545-4de3a2f4-8b88-4abd-b174-008877c56af2.png)
![Automation](https://user-images.githubusercontent.com/1493547/141676601-2f0cbef3-f79c-41e6-a5ce-cf31fdf4b9dd.png)

**Comment by section83** [comment]
Hello and sorry you’ve had this trouble.


> On 14 Nov 2021, at 9:28 pm, artcore-c ***@***.***> wrote:
> 
> Having some trouble getting MacYTDL off and running, keep getting this error message,
> 

The error is mine.  The YT-DLP developers could not build their latest version for macOS.  So, they left it out of the release.  I never thought that could happen and so, MacYTDL crashes.  The “Not” in the error refers to the content of YT-DLP that is installed.  It is a text file with “Not Found” inside.

I recommend you delete the YT-DLP file – you will need to provide Administrator credentials.  The older “youtube-dl” tool should still work – although it cannot download from YouTube faster than around 80kbps.  So, it might not be of much help to you.

There is also a bug in the “Auto check on startup” function so, leave it off.

I plan to release a new version later today (it’s 8.51am now AEDT).  It will fix those two bugs plus two bugs in downloading from SBS OnDemand.

> System Preferences/Privacy/Automation there isn't a checkbox for Safari?
> 

Probably will appear if you click on “OK” when/if a dialog appears.  It should appear when/if you start to use the “Service”.

>  It really looks like a nice UI what I see in the Help booklet, 
> 

Thanks.  It is a crude UI as I’ve written it almost entirely in AppleScript.  I hope one day to learn another language and redesign the UI to be more macOS-like.

> trying to replay the a few minutes over my slow mobile connection lags re-buffering and ends up taking a lot longer to see the parts I need.
> 

Yes.  Video downloaders are a great help when Internet access is slow or unreliable.

> I wish there was a way to download faster to help compensate for slow connectivity like I'm able to do with Download Shuttle
> 

Download speeds a normally a function of your Internet connection.  With the NBN some people are stuck with quite slow downloads.  I’ve not heard of Download Shuttle.  I’ll have a look.

Cheers.


Garry
***@***.***

**Reply by artcore-c** [reply]
Hi thanks for the prompt reply, had guessed maybe was something with yt-dlp. I found a solution in 4K downloader for Mac it's free under 30 downloads per day, well above my needs. I hadn't given up on this one yet though and will try the updates you mention. The next thing once I was able to download was native MacOS player QuickTime couldn't reduce playback speed so I'm very happy to report I discovered an open source project that also uses yt-dlp called IINA, it's a free media player with granular playback speed slider and many other tools like audio EQ. The UI looks just like QuickTime but with many added features in tab and side bars.

**Comment by section83** [comment]
> The next thing once I was able to download was native MacOS player QuickTime couldn't reduce playback speed

Thanks. I've not been aware if IINA. It seems to be highly regarded as an alternative to VLC, which I've been using for years.  VLC is very good but, maybe IINA is moving past it.  But, on my 12yo iMac, IINA makes some videos into stop motion while VLC plays smoothly.  So, maybe IINA is aimed at more recent hardware.

IINA comes with a copy of youtube-dl in its bundle, not YT-DLP. They really should mention youtube-dl in their credits. I hope there is a way to update/change the youtube-dl install to use YT-DLP. In the meantime, you can download _YT-DLP_, rename it to _youtube-dl_ then save it in folder used by IINA.

Cheers.

**Comment by section83** [comment]
I've released an update to MacYTDL (v1.19.1) which I hope solves this problem.
